src/solver/conv_mlir_igemm_fwd.cpp

```diff
@@ -110,18 +109,26 @@ bool PerformanceConvMlirIgemm::SetNextValue(const ConvolutionContext& /*config*/
     // always search full space, no matter if use_spare_set or not
     do
     {
-        if(!NextTwoPower<64, 256>(BlockSize))
-            break;
-        if(!NextTwoPower<32, 128>(GemmMPerBlock))
-            break;
-        if(!NextTwoPower<32, 128>(GemmNPerBlock))
-            break;
-        if(!NextTwoPower<4, 16>(GemmKPerBlock))
-            break;
-        if(!NextTwoPower<2, 4>(GemmMPerThread))
-            break;
-        if(!NextTwoPower<2, 4>(GemmNPerThread))
-            break;
+        if(use_spare_set)
+        {
+            if(!NextFlag<false, true>(SpareConfig))
+                break;
+        }
+        else
```

Comment:
Okay I think this is where I get myself confused. I somehow had this misunderstanding that generic search will discard the performance config when `SetNextValue()` return false. (Whereas actually `SetNextValue()` is really just to tell if the iterator has reached the end. It will try every step of the iteration.)

Let me get rid of the unnecessary complexity I added here.